int32 point_num
Point[] path_points
